lib/Dialect/FIRRTL/Transforms/InferReadWrite.cpp

```diff
@@ -72,7 +72,8 @@ struct InferReadWritePass : public InferReadWriteBase<InferReadWritePass> {
             // If this is the enable field, record the product terms(the And
             // expression tree).
             if (fName.equals("en"))
-              getProductTerms(sf, portTerms[portIt.index()]);
+              getProductTerms(sf, portTerms[readPort ? 0 : 1]);
```

Comment:
The `portTerms` 2D vec makes some of this logic confusing. It may be cleaner to split up `portTerms[2]` into `readPortTerms` and `writePortTerms` to avoid the index convention. This is somewhat out of scope of this PR, though.

If not changed (in scope of this PR), this `readPort ? 0 : 1` can just be `!readPort`. One thing that is slightly confusing here is the name of the variable as `readPort` when it's really `isReadPort`.

Reply:
Thanks Schuyler, I had the same thoughts, updated the implementation to use `readTerms` and `writeTerms` and `isReadPort`.

Reply:
Looks way better! Thanks!